src/main/java/com/meilisearch/sdk/TasksHandler.java

```diff
@@ -26,53 +28,93 @@ public class TasksHandler {
+     * Retrieves tasks from the client
      *
-     * @param indexUid Index identifier to the requested Tasks
      * @return List of task instance
      * @throws MeilisearchException if client request causes an error
      */
-    Results<Task> getTasks(String indexUid) throws MeilisearchException {
+    TasksResults getTasks() throws MeilisearchException {
+        String urlPath = "/tasks";
 
-        Results<Task> result = httpClient.get(urlPath, Results.class, Task.class);
+        TasksResults result = httpClient.get(urlPath, TasksResults.class);
         return result;
     }
 
     /**
-     * Retrieves the task with the specified task uid
+     * Retrieves tasks from the client
      *
-     * @param taskUid Identifier of the requested Task
-     * @return Task instance
+     * @param param accept by the tasks route
+     * @return List of task instance
      * @throws MeilisearchException if client request causes an error
      */
-    Task getTask(int taskUid) throws MeilisearchException {
-        String urlPath = "/tasks/" + taskUid;
-        return httpClient.get(urlPath, Task.class);
+    TasksResults getTasks(TasksQuery param) throws MeilisearchException {
+        URLBuilder urlb = new URLBuilder();
+        urlb.addSubroute("tasks");
+        urlb.addParameter("limit", param.getLimit());
+        urlb.addParameter("from", param.getFrom());
+        urlb.addParameter("status", param.getStatus());
+        urlb.addParameter("type", param.getType());
+        urlb.addParameter("indexUid", param.getIndexUid());
+        String urlQuery = urlb.getURL();
+
+        TasksResults result = httpClient.get(urlQuery, TasksResults.class);
+        return result;
     }
 
     /**
-     * Retrieves tasks from the client
+     * Retrieves all TasksHandler at the specified index uid
      *
+     * @param indexUid Index identifier to the requested Tasks
      * @return List of task instance
      * @throws MeilisearchException if client request causes an error
      */
-    Results<Task> getTasks() throws MeilisearchException {
-        String urlPath = "/tasks";
+    TasksResults getTasks(String indexUid) throws MeilisearchException {
+        URLBuilder urlb = new URLBuilder();
+        urlb.addSubroute("tasks");
+        urlb.addParameter("indexUid", indexUid);
+        String urlQuery = urlb.getURL();
+
+        TasksResults result = httpClient.get(urlQuery, TasksResults.class);
+        return result;
+    }
+
+    /**
+     * Retrieves all TasksHandler at the specified index uid
+     *
+     * @param indexUid Index identifier to the requested Tasks
+     * @param param accept by the tasks route
+     * @return List of task instance
+     * @throws MeilisearchException if client request causes an error
+     */
+    TasksResults getTasks(String indexUid, TasksQuery param) throws MeilisearchException {
+        String[] newIndexUid = new String[param.getIndexUid().length + 1];
```

Comment:
I'm not sure why you had to do this. Can you explain it to me, please?

Reply:
It's for this specific case:
```java
    TasksQuery query = new TasksQuery().setIndexes(new String[] {"books", "cities"});
    client.index("movies").getTasks();
```
In case the `getTasks()` method is called through the `IndexesHandler` the request of this should be:
```
/tasks?indexUids=movies,books,cities
```